inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcBatchingOutputFormat.java

```diff
@@ -0,0 +1,463 @@
+        return JdbcBatchStatementExecutor.simple(
+                sql,
+                createRowJdbcStatementBuilder(fieldTypes),
+                objectReuse ? Row::copy : Function.identity());
+    }
+
+    /**
+     * Creates a {@link JdbcStatementBuilder} for {@link Row} using the provided SQL types array.
+     * Uses {@link JdbcUtils#setRecordToStatement}
+     */
+    static JdbcStatementBuilder<Row> createRowJdbcStatementBuilder(int[] types) {
+        return (st, record) -> setRecordToStatement(st, types, record);
+    }
+
+    /**
+     * Connects to the target database and initializes the prepared statement.
+     *
+     * @param taskNumber The number of the parallel instance.
+     */
+    @Override
+    public void open(int taskNumber, int numTasks) throws IOException {
+        super.open(taskNumber, numTasks);
+        this.runtimeContext = getRuntimeContext();
+        metricData = new MetricData(runtimeContext.getMetricGroup());
+        if (inLongMetric != null && !inLongMetric.isEmpty()) {
+            String[] inLongMetricArray = inLongMetric.split("_");
+            String groupId = inLongMetricArray[0];
+            String streamId = inLongMetricArray[1];
+            String nodeId = inLongMetricArray[2];
+            metricData.registerMetricsForDirtyBytes(groupId, streamId, nodeId, "dirtyBytes");
+            metricData.registerMetricsForDirtyRecords(groupId, streamId, nodeId, "dirtyRecords");
+            metricData.registerMetricsForNumBytesOut(groupId, streamId, nodeId, "numBytesOut");
+            metricData.registerMetricsForNumRecordsOut(groupId, streamId, nodeId, "numRecordsOut");
+            metricData.registerMetricsForNumBytesOutPerSecond(groupId, streamId, nodeId, "numBytesOutPerSecond");
+            metricData.registerMetricsForNumRecordsOutPerSecond(groupId, streamId, nodeId,
+                    "numRecordsOutPerSecond");
+        }
+        jdbcStatementExecutor = createAndOpenStatementExecutor(statementExecutorFactory);
+        if (executionOptions.getBatchIntervalMs() != 0 && executionOptions.getBatchSize() != 1) {
+            this.scheduler =
+                    Executors.newScheduledThreadPool(
+                            1, new ExecutorThreadFactory("jdbc-upsert-output-format"));
+            this.scheduledFuture =
+                    this.scheduler.scheduleWithFixedDelay(
+                            () -> {
+                                synchronized (JdbcBatchingOutputFormat.this) {
+                                    if (!closed) {
+                                        try {
+                                            flush();
+                                            if (metricData.getNumRecordsOut() != null) {
+                                                metricData.getNumRecordsOut().inc(rowSize);
+                                            }
+                                            if (metricData.getNumRecordsOut() != null) {
```

Comment:
`metricData.getNumRecordsOut() != null` should be `metricData.getNumBytesOut()`

Reply:
OK